src/BuildConfig.cc

```diff
@@ -817,8 +852,16 @@ BuildConfig::configureBuild() {
 
   setVariables();
 
+  std::unordered_set<std::string> buildAll = {};
+  if (executable) {
+    buildAll.insert(packageName);
+  }
+  if (library) {
+    buildAll.insert(fmt::format("lib{}.a", packageName));
```

Comment:
I went ahead and made a `libName` member in `BuildConfig`. I also made a `getLibraryName` function in `Manifest.cpp`. It made me realize that the code mixes and matches them very liberally, so I'm not sure what the convention would be here. Ideally, we only have one call to `fmt::format("lib{}.a", packageName)`, but I was only able to pare it down to two calls. I'm not sure what the best solution is.

Reply:
Is `libName` used in other methods? If so, your change makes sense.

As for `BuildConfig.packageName` and `getPackageName()`, yes I agree. But as this line `BuildConfig config(getPackageName(), isDebug);` in `emitMakefile` shows, they are the same. We can consider always calling `getPackageName()` instead of storing it in `BuildConfig.packageName` or another idea, but not now.

I think my suggestion below resolves this issue, `fmt::format("lib{}.a", packageName)`.

Reply:
Sounds good!